client/src/components/SearchResultsOffset.js

```diff
@@ -0,0 +1,166 @@
+import React from 'react'
+import { Box, Button, Paragraph, TextInput } from 'grommet'
+import { FormPrevious, FormNext } from 'grommet-icons'
+import { isOnlyNumbers } from '../helpers/isOnlyNumbers'
+import { useSearchResources } from '../hooks/useSearchResources'
+
+export const SearchResultsOffset = () => {
+  const {
+    query: { limit, offset },
+    response: { count },
+    setOffset,
+    goToSearchResults
+  } = useSearchResources()
+
+  const [enteredPageNumber, setEnteredPageNumber] = React.useState('')
+  const [
+    enteredPageNumberInRange,
+    setEnteredPageNumberInRange
+  ] = React.useState(false)
+
+  const safeOffset = parseInt(offset, 10) || 0
+  const safeCount = parseInt(count, 10) || 0
+  const parsedLimit = parseInt(limit, 10)
+  const lastOffset =
+    safeCount && parsedLimit > 0 ? Math.floor(safeCount / parsedLimit) : 0
+
+  const startingPageSlices =
+    safeOffset <= 2 ? [0, 6] : [safeOffset - 2, safeOffset + 3]
+
+  const pageSlices = lastOffset - safeOffset <= 2 ? [-5] : startingPageSlices
+
+  const pageButtonsOffsets = [...Array(lastOffset + 1).keys()].slice(
+    ...pageSlices
+  )
+
+  const goToOffset = (page) => {
+    setOffset(page)
+    setEnteredPageNumber('')
+    goToSearchResults()
```

Comment:
If `page` is the same as the current offset, will `goToSearchResults()` actually do anything or will it recognize that nothing has changed?

Reply:
it basically refreshes the page